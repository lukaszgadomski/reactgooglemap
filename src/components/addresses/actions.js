import actionTypes from "./actionTypes";

export const clickAddress = address => {
  return {
    type: actionTypes.CLICK_ADDRESS,
    address
  };
};

export const fetchAddressRequest = () => {
  return {
    type: actionTypes.FETCH_REQUEST_ADDRESS,
    fetching: true
  };
};

export const fetchAddressError = errMsg => {
  return {
    type: actionTypes.FETCH_ERROR_ADDRESS,
    fetching: false,
    err: errMsg
  };
};

export const fetchAddressSuccess = res => {
  return {
    type: actionTypes.FETCH_SUCCESS_ADDRESS,
    fetching: false,
    addresses: res.addresses,
    markers: res.markers
  };
};
