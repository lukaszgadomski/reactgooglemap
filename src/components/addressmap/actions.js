import actionTypes from "./actionTypes";

export const clickAddress = address => {
  return {
    type: actionTypes.CLICK_ADDRESS,
    address
  };
};
