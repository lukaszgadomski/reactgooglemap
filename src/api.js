import config from "./config";
import * as actions from "./components/addresses/actions";
import axios from "axios";

export const dayOrder = {
  Sunday: "Niedziela",
  Monday: "Poniedziałek",
  Tuesday: "Wtorek",
  Wednesday: "Środa",
  Thursday: "Czwartek",
  Friday: "Piątek",
  Saturday: "Sobota"
};
export const cleanupAddressData = addresses => {
  let id = 1;
  return addresses.reduce(
    (ret, a) => {
      ret.addresses.push({
        id,
        address: a.adres,
        hours: Object.keys(dayOrder).map(day => {
          return {
            day: dayOrder[day],
            hours: (a.godziny && a.godziny[0] && a.godziny[0][day]) ||
              "Brak danych"
          };
        }),
        name: a.siec,
        lat: a.latitude,
        lng: a.longitude,
        icon: a.logo
      });
      ret.markers.push({
        id,
        icon: a.logo,
        lat: a.latitude,
        lng: a.longitude
      });
      id++;
      return ret;
    },
    {
      addresses: [],
      markers: []
    }
  );
};

export const apiAddress = {
  fetchPromise: function() {
    return axios.get(`${config.ADDRESS_SOURCE_URL}`);
  },
  fetch: function(store) {
    store.dispatch(actions.fetchAddressRequest());
    return this.fetchPromise()
      .then(r => r.data)
      .then(json => {
        store.dispatch(actions.fetchAddressSuccess(cleanupAddressData(json)));
        return json;
      })
      .catch(function(error) {
        store.dispatch(actions.fetchAddressError(error));
      });
  }
};
