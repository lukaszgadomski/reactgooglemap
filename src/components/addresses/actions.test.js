import * as actions from "./actions";
import types from "./ActionTypes";

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { apiAddress } from "../../api";
import sinon from "sinon";

const rawAddressData = [
  {
    adres: "Warszawa, ul. Grzybowska 37",
    godziny: [
      {
        Friday: "10:00 - 19:00",
        Monday: "10:00 - 19:00",
        Saturday: "brak danych",
        Sunday: "brak danych",
        Thursday: "10:00 - 19:00",
        Tuesday: "10:00 - 19:00",
        Wednesday: "10:00 - 19:00"
      }
    ],
    latitude: "52.2351646",
    logo: "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/6/s3_logo_alsen_siec-handlowa.jpg",
    longitude: "20.9942039",
    siec: "Alsen"
  },
  {
    adres: "Warszawa, al. Jana Pawła II 52/54",
    godziny: [
      {
        Friday: "24h",
        Monday: "24h",
        Saturday: "24h",
        Sunday: "24h",
        Thursday: "24h",
        Tuesday: "24h",
        Wednesday: "24h"
      }
    ],
    latitude: "52.2446115",
    logo: "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/120/s3_logo_apteka1_siec-handlowa.jpg",
    longitude: "20.9919215",
    siec: "Apteka1"
  }
];

const processedData = {
  addresses: [
    {
      id: 1,
      address: "Warszawa, ul. Grzybowska 37",
      hours: [
        { day: "Niedziela", hours: "brak danych" },
        { day: "Poniedziałek", hours: "10:00 - 19:00" },
        { day: "Wtorek", hours: "10:00 - 19:00" },
        { day: "Środa", hours: "10:00 - 19:00" },
        { day: "Czwartek", hours: "10:00 - 19:00" },
        { day: "Piątek", hours: "10:00 - 19:00" },
        { day: "Sobota", hours: "brak danych" }
      ],
      name: "Alsen",
      lat: "52.2351646",
      lng: "20.9942039"
    },
    {
      id: 2,
      address: "Warszawa, al. Jana Pawła II 52/54",
      hours: [
        { day: "Niedziela", hours: "24h" },
        { day: "Poniedziałek", hours: "24h" },
        { day: "Wtorek", hours: "24h" },
        { day: "Środa", hours: "24h" },
        { day: "Czwartek", hours: "24h" },
        { day: "Piątek", hours: "24h" },
        { day: "Sobota", hours: "24h" }
      ],
      name: "Apteka1",
      lat: "52.2446115",
      lng: "20.9919215"
    }
  ],
  markers: [
    {
      id: 1,
      icon: "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/6/s3_logo_alsen_siec-handlowa.jpg",
      lat: "52.2351646",
      lng: "20.9942039"
    },
    {
      id: 2,
      icon: "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/120/s3_logo_apteka1_siec-handlowa.jpg",
      lat: "52.2446115",
      lng: "20.9919215"
    }
  ]
};
describe("synchronous map actions", () => {
  it("should create an click action to center map and highlight address on list", () => {
    const given = {
      id: 2,
      address: "Warszawa, al. Jana Pawła II 52/54",
      hours: [
        { day: "Niedziela", hours: "24h" },
        { day: "Poniedziałek", hours: "24h" },
        { day: "Wtorek", hours: "24h" },
        { day: "Środa", hours: "24h" },
        { day: "Czwartek", hours: "24h" },
        { day: "Piątek", hours: "24h" },
        { day: "Sobota", hours: "24h" }
      ],
      name: "Apteka1",
      lat: "52.2446115",
      lng: "20.9919215"
    };

    expect(actions.clickAddress(given)).toMatchSnapshot();
  });

  it("should create action fetch address", () => {
    const given = {
      type: types.FETCH_REQUEST_ADDRESS
    };
    expect(actions.fetchAddressRequest()).toMatchSnapshot();
  });

  it("should create action fetch error address", () => {
    const givenErr = "Timeout";
    const expected = {
      type: types.FETCH_ERROR_ADDRESS,
      fetching: false,
      err: "Timeout"
    };
    expect(actions.fetchAddressError(givenErr)).toMatchSnapshot();
  });

  it("should create action fetch success address", () => {
    const given = processedData;
    const expected = {
      type: types.FETCH_SUCCESS_ADDRESS,
      fetching: false,
      addresses: given.addresses,
      markers: given.markers
    };
    expect(actions.fetchAddressSuccess(given)).toMatchSnapshot();
  });
});

describe("async fetch actions", () => {
  it(`creates correct actions on address api fetch`, () => {
    const given = processedData;
    const expectedActions = [
      {
        type: types.FETCH_REQUEST_ADDRESS,
        fetching: true
      },
      {
        type: types.FETCH_SUCCESS_ADDRESS,
        fetching: false,
        addresses: given.addresses,
        markers: given.markers
      }
    ];
    const mockStore = configureStore([]);
    const store = mockStore({});

    const callback = sinon
      .stub(apiAddress, "fetchPromise")
      .callsFake(function fakeFn() {
        return Promise.resolve({ data: rawAddressData });
      });

    return apiAddress.fetch(store).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
