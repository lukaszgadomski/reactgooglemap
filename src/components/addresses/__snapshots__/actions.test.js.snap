// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async fetch actions creates correct actions on address api fetch 1`] = `
Array [
  Object {
    "fetching": true,
    "type": "FETCH_REQUEST_ADDRESS",
  },
  Object {
    "addresses": Array [
      Object {
        "address": "Warszawa, ul. Grzybowska 37",
        "hours": Array [
          Object {
            "day": "Niedziela",
            "hours": "brak danych",
          },
          Object {
            "day": "Poniedziałek",
            "hours": "10:00 - 19:00",
          },
          Object {
            "day": "Wtorek",
            "hours": "10:00 - 19:00",
          },
          Object {
            "day": "Środa",
            "hours": "10:00 - 19:00",
          },
          Object {
            "day": "Czwartek",
            "hours": "10:00 - 19:00",
          },
          Object {
            "day": "Piątek",
            "hours": "10:00 - 19:00",
          },
          Object {
            "day": "Sobota",
            "hours": "brak danych",
          },
        ],
        "icon": "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/6/s3_logo_alsen_siec-handlowa.jpg",
        "id": 1,
        "lat": "52.2351646",
        "lng": "20.9942039",
        "name": "Alsen",
      },
      Object {
        "address": "Warszawa, al. Jana Pawła II 52/54",
        "hours": Array [
          Object {
            "day": "Niedziela",
            "hours": "24h",
          },
          Object {
            "day": "Poniedziałek",
            "hours": "24h",
          },
          Object {
            "day": "Wtorek",
            "hours": "24h",
          },
          Object {
            "day": "Środa",
            "hours": "24h",
          },
          Object {
            "day": "Czwartek",
            "hours": "24h",
          },
          Object {
            "day": "Piątek",
            "hours": "24h",
          },
          Object {
            "day": "Sobota",
            "hours": "24h",
          },
        ],
        "icon": "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/120/s3_logo_apteka1_siec-handlowa.jpg",
        "id": 2,
        "lat": "52.2446115",
        "lng": "20.9919215",
        "name": "Apteka1",
      },
    ],
    "fetching": false,
    "markers": Array [
      Object {
        "icon": "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/6/s3_logo_alsen_siec-handlowa.jpg",
        "id": 1,
        "lat": "52.2351646",
        "lng": "20.9942039",
      },
      Object {
        "icon": "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/120/s3_logo_apteka1_siec-handlowa.jpg",
        "id": 2,
        "lat": "52.2446115",
        "lng": "20.9919215",
      },
    ],
    "type": "FETCH_SUCCESS_ADDRESS",
  },
]
`;

exports[`synchronous map actions should create action fetch address 1`] = `
Object {
  "fetching": true,
  "type": "FETCH_REQUEST_ADDRESS",
}
`;

exports[`synchronous map actions should create action fetch error address 1`] = `
Object {
  "err": "Timeout",
  "fetching": false,
  "type": "FETCH_ERROR_ADDRESS",
}
`;

exports[`synchronous map actions should create action fetch success address 1`] = `
Object {
  "addresses": Array [
    Object {
      "address": "Warszawa, ul. Grzybowska 37",
      "hours": Array [
        Object {
          "day": "Niedziela",
          "hours": "brak danych",
        },
        Object {
          "day": "Poniedziałek",
          "hours": "10:00 - 19:00",
        },
        Object {
          "day": "Wtorek",
          "hours": "10:00 - 19:00",
        },
        Object {
          "day": "Środa",
          "hours": "10:00 - 19:00",
        },
        Object {
          "day": "Czwartek",
          "hours": "10:00 - 19:00",
        },
        Object {
          "day": "Piątek",
          "hours": "10:00 - 19:00",
        },
        Object {
          "day": "Sobota",
          "hours": "brak danych",
        },
      ],
      "id": 1,
      "lat": "52.2351646",
      "lng": "20.9942039",
      "name": "Alsen",
    },
    Object {
      "address": "Warszawa, al. Jana Pawła II 52/54",
      "hours": Array [
        Object {
          "day": "Niedziela",
          "hours": "24h",
        },
        Object {
          "day": "Poniedziałek",
          "hours": "24h",
        },
        Object {
          "day": "Wtorek",
          "hours": "24h",
        },
        Object {
          "day": "Środa",
          "hours": "24h",
        },
        Object {
          "day": "Czwartek",
          "hours": "24h",
        },
        Object {
          "day": "Piątek",
          "hours": "24h",
        },
        Object {
          "day": "Sobota",
          "hours": "24h",
        },
      ],
      "id": 2,
      "lat": "52.2446115",
      "lng": "20.9919215",
      "name": "Apteka1",
    },
  ],
  "fetching": false,
  "markers": Array [
    Object {
      "icon": "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/6/s3_logo_alsen_siec-handlowa.jpg",
      "id": 1,
      "lat": "52.2351646",
      "lng": "20.9942039",
    },
    Object {
      "icon": "https://static.okazjum.pl/storage/okazjum/uploads/uploads/contractor/logo/120/s3_logo_apteka1_siec-handlowa.jpg",
      "id": 2,
      "lat": "52.2446115",
      "lng": "20.9919215",
    },
  ],
  "type": "FETCH_SUCCESS_ADDRESS",
}
`;

exports[`synchronous map actions should create an click action to center map and highlight address on list 1`] = `
Object {
  "address": Object {
    "address": "Warszawa, al. Jana Pawła II 52/54",
    "hours": Array [
      Object {
        "day": "Niedziela",
        "hours": "24h",
      },
      Object {
        "day": "Poniedziałek",
        "hours": "24h",
      },
      Object {
        "day": "Wtorek",
        "hours": "24h",
      },
      Object {
        "day": "Środa",
        "hours": "24h",
      },
      Object {
        "day": "Czwartek",
        "hours": "24h",
      },
      Object {
        "day": "Piątek",
        "hours": "24h",
      },
      Object {
        "day": "Sobota",
        "hours": "24h",
      },
    ],
    "id": 2,
    "lat": "52.2446115",
    "lng": "20.9919215",
    "name": "Apteka1",
  },
  "type": "CLICK_ADDRESS",
}
`;
